mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.purgeOrder = async (req, res) => {
+  try {
+    const result = await Order.findOneAndRemove({ _id: req.params.orderId });
+    if (result) {
+      return res.status(404).json({ message: "Order not located" });
```

Comment:
In the `purgeOrder` function, the 404 status code is returned when the order is found and removed, which is counterintuitive. It should be returned when the order is not found. The condition should be `if (!result) { ... }` instead of `if (result) { ... }`.